### ISSUE TITLE
Ignore escaped separators

### ISSUE THREAD

**Issue by gausie** [elgs/splitargs#3]
If my separator is " ", I'd like "\ " not to be considered a separator.

**Comment by elgs** [comment]
I don't think it works that way. Anything inside the quotes, either single or double, will not be considered as a separator. The only things need to be escaped are the single quote ' and the double quote ". If you don't want some of the white spaces being considered as separator, why not put them in a pair of quotes?

**Comment by gausie** [comment]
I'm in a situation where I cannot control the input, unfortunately. Escaped
spaces are accepted in shell and I'm trying to parse shell arguments.

**Comment by elgs** [comment]
I see. Let me think about how to implement it.

**Comment by greggman** [comment]
Not sure if this works but here's my attempt

http://jsfiddle.net/greggman/c69au3oh/

**Comment by elgs** [comment]
Thanks @greggman. The results look pretty good. I will do more test.

**Comment by Flamenco** [comment]
I needed to escape spaces outside of my quotes, and also quotes inside my quotes.

Single Argument
hello\ world 

Quotes of same type inside.
'hello\\'world\\'' 

See PR: https://github.com/elgs/splitargs/pull/6
